tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,794 @@
+        self.logger.info(
+            f"calculated hwm: {stats['hwm']}, message count: {self.params.msg_count}"
+        )
+        chk.add_arg("sum_high_watermarks", stats['hwm'])
+        chk.add_arg("msg_count", self.params.msg_count)
+        chk.add_check(
+            "Ensure that all messages made it to topic",
+            "sum_high_watermarks >= msg_count",
+        )
+        # Local timestamp for latest message
+        self.logger.info(f"Max local ts = {stats['local_ts']}")
+
+        # Measure how far behind the tiered storage uploads are:
+        # success condition should be that they are within some
+        # time range of the most recently produced data
+        if not stats['uploaded_ts']:
+            self.logger.warn(
+                "No uploaded segments found! Check test parameters!")
+        else:
+            self.logger.info(f"Max uploaded ts = {stats['uploaded_ts']}")
+
+            lag_seconds = (stats['local_ts'] - stats['uploaded_ts']) / 1000.0
+            config_interval = (self.params.manifest_upload_interval +
+                               (self.params.segment_size / actual_byte_rate))
+            self.logger.info(f"Upload lag: {lag_seconds}s, "
+                             f"Upload interval: {config_interval}")
+            chk.add_arg("lag_seconds", lag_seconds)
+            chk.add_arg("config_interval", config_interval)
+            chk.add_check(
+                "Measure how far behind the tiered storage uploads are: "
+                "success condition should be that they are within some "
+                "time range of the most recently produced data",
+                "lag_seconds < config_interval",
+            )
+
+        # Wait for all uploads to complete: this should take roughly
+        # segment_max_upload_interval_sec plus manifest_max_upload_interval_sec
+        quiesce_uploads(self.redpanda, [self.topic],
+                        timeout_sec=self.params.manifest_upload_interval +
+                        self.params.segment_upload_interval)
+
+        # Check manifest upload metrics:
+        #  - we should not have uploaded the manifest more times
+        #    then there were manifest upload intervals in the runtime.
+        self.logger.info(
+            f"Upload counts: {stats['manifest_uploads']} manifests, "
+            f"{stats['segment_uploads']} segments")
+        chk.add_arg("manifest_uploads", stats['manifest_uploads'])
+        chk.add_arg("segment_uploads", stats['segment_uploads'])
+        chk.add_check(
+            "Non-zero manifest uploads",
+            "manifest_uploads > 0",
+        )
+        chk.add_check(
+            "Non-zero segment uploads",
+            "segment_uploads > 0",
+        )
+
+        # +3 because:
+        # - 1 empty upload at start
+        # - 1 extra upload from runtime % upload interval
+        # - 1 extra upload after the final interval_sec driven uploads
+        expect_manifest_uploads = (
```

Comment:
This is calculated but not used. I'm not sure you really need it in this test though.

Reply:
This is a part of old assertion that was skipped after discussion as non-critical in this test. Will remove.